src/main/java/org/jvnet/hudson/test/RealJenkinsRule.java

```diff
@@ -457,6 +462,127 @@ public RealJenkinsRule prepareHomeLazily(boolean prepareHomeLazily) {
+            throw new RuntimeException(e);
+        }
+        return this;
+    }
+
+    /**
+     *
+     * @param files add some {@link File} to bootclasspath
+     */
+    public RealJenkinsRule withBootClasspath(File...files) {
+        this.bootClasspathFiles.addAll(List.of(files));
+        return this;
+    }
+
+    private IllegalArgumentException fipsIllegalSetupException(Path fipsLibrariesPath) {
+        String message = "Path " + fipsLibrariesPath.toString() + " must contains the FIPS BouncyCastle jars:";
+        message += "bc-fips.jar, bctls-fips.jar and bcpkix-fips.jar" + System.lineSeparator();
+        message += "Configure your pom.xml with this plugin execution: " + System.lineSeparator();
+        message += "<plugin>\n" +
+                "  <groupId>org.apache.maven.plugins</groupId>\n" +
+                "  <artifactId>maven-dependency-plugin</artifactId>\n" +
+                "  <executions>\n" +
+                "    <execution>\n" +
+                "      <id>copy-libs</id>\n" +
+                "      <goals>\n" +
+                "        <goal>copy</goal>\n" +
+                "      </goals>\n" +
+                "      <configuration>\n" +
+                "        <artifactItems>\n" +
+                "          <artifactItem>\n" +
+                "            <groupId>org.bouncycastle</groupId>\n" +
+                "            <artifactId>bc-fips</artifactId>\n" +
+                "            <version>1.0.2.4</version>\n" +
+                "          </artifactItem>\n" +
+                "          <artifactItem>\n" +
+                "            <groupId>org.bouncycastle</groupId>\n" +
+                "            <artifactId>bcpkix-fips</artifactId>\n" +
+                "            <version>1.0.7</version>\n" +
+                "          </artifactItem>\n" +
+                "          <artifactItem>\n" +
+                "            <groupId>org.bouncycastle</groupId>\n" +
+                "            <artifactId>bctls-fips</artifactId>\n" +
+                "            <version>1.0.17</version>\n" +
+                "          </artifactItem>\n" +
+                "        </artifactItems>\n" +
+                "        <stripVersion>true</stripVersion>\n" +
+                "      </configuration>\n" +
+                "    </execution>\n" +
+                "  </executions>\n" +
+                "</plugin>";
+        return new IllegalArgumentException(message);
+    }
+
+    private Path writeFIPSJavaSecurityFile() throws IOException {
+        String javaHome = System.getProperty("java.home");
+        if(javaHome == null) {
+            throw new IllegalArgumentException("Cannot find java.home property");
+        }
+        Path javaSecurity = Paths.get(javaHome, "conf", "security", "java.security");
+        Properties properties = new Properties();
+        Path securityFile = Files.createTempFile("java", ".security");
+        securityFile.toFile().deleteOnExit();
```

Comment:
NIT: There is actually a `TemporaryDirectoryAllocator tmp`  (although it is currently only cleans up if home is `non-null` but that could be changed?)